src/czt.jl

```diff
@@ -1,107 +1,305 @@
+export czt, iczt, plan_czt, CZTPlan_ND
+
+"""
+    get_kernel_1d(RT::Type, N::Integer, M::Integer; a= 1.0, w = cispi(-2/N), extra_phase=0.0, global_phase=0.0)
+
+calculates the kernel for the Bluestein algorithm. Note that the length depends on the destination size.
+Note the the resulting kernel-size is computed based on the minimum required length for the task.
+Use `size(result)` to know what to calculate with. The center of the resulting `kernel` is set to the standard
+convention position: 0 
+The code is based on Rabiner, Schafer & Rader  1969, IEEE Trans. on Audio and Electroacoustics, 17,86-92
+
+# Arguments
+    `RT`: the datatype of the array to transform
+    `N`: an integer for the length of the source array to transform.
+    `M`: an integer for the length of the destination array.
+    `a`: the (complex valued) phasor defining the start of the sampling in the source array
+    `w`: the (complex valued) phasor defining consecutive sample positions
+    `extra_phase`: a consecutive phase to apply to the final result, which enables to change the interpretation of the central source positon. 
+    `global_phase`: the start phase of the `extra_phase` to apply.
+    returns: a tuple of three arrays for the initial multiplication (A*W), the convolution (already fourier-transformed) and the post multiplication.
+"""
+function get_kernel_1d(RT::Type, N::Integer, M::Integer; a= 1.0, w = cispi(-2/N), extra_phase=0.0, global_phase=0.0)
+    # intorduce also sscale ??
+    # the size needed to avoid wrap
+    CT = (RT <: Real) ? Complex{RT} : RT
+    RT = real(CT)
+    # nowrap_size = N + ceil(N÷2)
+    # the maximal size where the convolution does not yield zero 
+    # max_size = 2*N-1
+    # the minimum size needed for the convolution
+    # kernel-size for the deconvolution
+    L = N + M -1 # max(min(max_size, dsize), nowrap_size)
+    
+    #Note that the source size ssz is used here
+     # W of the source for FFTs. 
+    n = (0:N-1)
+    # pre-calculate the product of a.^(-n) and w to be later multiplied with the input x
+    # late casting is important, since the rounding errors are overly large if the original calculations are done in Float32.
+    aw = CT.((a .^ (-n)) .* w .^ ((n .^ 2) ./ 2))
+
+    conv_kernel = zeros(CT, L) # Array{CT}(undef, L)
+    m = (0:M-1)
+    conv_kernel[1:M] .= w .^ (-(m .^ 2) ./ 2)
+    right_start = L-N+1
+    n = (1:N-1)
+    conv_kernel[L:-1:right_start+1] .= w .^ (.-(n .^ 2) ./ 2)
+    # calculate W^(k^2/2), we can reuse the conv_kernel definition 1..M and just invert it 
+    #
+    # The extra_phase accounts for the nominal center position of the source array. Note that this is 
+    # by default selected to be the geometric center and not the Fourier-center result in a real array upon zooming a symmetric even array.
+    wd = (abs(w)≈1.0) ? CT.(conj.(conv_kernel[1:M]) .* global_phase .* extra_phase.^m) : CT.(inv.(conv_kernel[1:M]) .* global_phase .* extra_phase.^m)
```

Comment:
Why is `CT.` needed here?
Can't you assure the types of `global_phase` before such that the promotion is already correct.
Prevents a copy.

Reply:
I tried that first, but discovered that the calculation precision is inacceptable if originally done with the low-precision datatype. I decided that this 1D overhead is acceptable in the general case. Therefore also the default type of `global_phase` is not forced to the default datatype.
E.g.:
```julia
julia> FourierTools.get_kernel_1d(Float32, 1000, 1000, w=ComplexF64(cispi(-2/100)), extra_phase=00.0, global_phase=0.0)[2][2]
0.9298623f0 - 0.070280485f0im

julia> FourierTools.get_kernel_1d(Float32, 1000, 1000, w=ComplexF32(cispi(-2/100)), extra_phase=0f0, global_phase=0f0)[2][2]
0.94716537f0 - 0.079703294f0im
```